cupy/lib/_routines_poly.py

```diff
@@ -114,6 +115,121 @@ def _polypow(x, n):
     return cupy.convolve(x, _polypow(cupy.convolve(x, x), (n - 1) // 2))
 
 
+def _polyfit_typecast(x):
+    if x.dtype.kind == 'c':
+        return x.astype(numpy.complex128, copy=False)
+    return x.astype(numpy.float64, copy=False)
+
+
+def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
+    """Returns the least squares fit of polynomial of degree deg
+    to the data y sampled at x.
+
+    Args:
+        x (cupy.ndarray or cupy.poly1d): x-coordinates of the sample
```

Comment:
The inputs are not polynomial coefficients, but x/y-coordinates of the sample points. So, poly1d object should not be supported.